File: examquiz-ai/frontend/src/services/authService.js
import { apiClient } from './apiClient';

export const authService = {
  async register({ fullName, email, password }) {
    const response = await apiClient.post('/api/v1/auth/register', { fullName, email, password });
    return response.data.data; // AuthResponse
  },

  async login({ email, password }) {
    const response = await apiClient.post('/api/v1/auth/login', { email, password });
    return response.data.data; // AuthResponse
  },

  async refresh(refreshToken) {
    const response = await apiClient.post('/api/v1/auth/refresh', { refreshToken });
    return response.data.data; // AuthResponse
  },
};

File: examquiz-backend/src/main/java/com/examquizai/backend/service/AuthService.java
package com.examquizai.backend.service;

import com.examquizai.backend.dto.request.LoginRequest;
import com.examquizai.backend.dto.request.RefreshTokenRequest;
import com.examquizai.backend.dto.request.RegisterRequest;
import com.examquizai.backend.dto.response.AuthResponse;

/**
 * Handles user registration, login, and token refresh.
 */
public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    AuthResponse refreshToken(RefreshTokenRequest request);
}
